src/features/packages/pack/nugetPackageBuilder.ts

```diff
@@ -0,0 +1,48 @@
+import { PackArgs } from "./packArgs";
+import { doZip } from "./zipUtils";
+import fs from "fs";
+import os from "os";
+import path from "path";
+
+type NuSpecArgs = {
+    description: string;
+    authors: string[];
+    releaseNotes?: string;
+};
+
+type NuGetPackArgs = {
+    nuspecArgs?: NuSpecArgs;
+} & PackArgs;
+
+export class NuGetPackageBuilder {
+    async pack(args: NuGetPackArgs): Promise<string> {
+        const archiveFilename = `${args.packageId}.${args.version}.nupkg`;
+        const tmpFolder = os.tmpdir();
+        const inputFilePatterns = args.inputFilePatterns;
+
+        if (args.nuspecArgs) {
+            const nuspecFile = path.join(tmpFolder, `${args.packageId}.nuspec`);
+            fs.writeFileSync(nuspecFile, '<?xml version="1.0" encoding="utf-8"?>\n');
+            fs.appendFileSync(nuspecFile, '<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">\n');

```

Comment:
Is this the latest version? Are there other versions we should consider?

Reply:
I got this from https://learn.microsoft.com/en-us/nuget/reference/nuspec#general-form-and-schema, so pretty sure it's latest.